### ISSUE TITLE
[Python] Segmentation Fault on 22e4217a21311b5af94a166366395209ac9434b7d5cf529045d831ce904cf38d

### ISSUE THREAD

**Issue by gdesmar** [lief-project/LIEF#631]
**Describe the bug**
LIEF generates a segmentation fault when accessing the icons of the resources_manager.

**To Reproduce**
```
Python 3.9.7 (default, Sep 16 2021, 13:09:58)
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import lief
>>> lief.__version__
'0.11.5-37bc2c9'
>>> b = lief.parse("22e4217a21311b5af94a166366395209ac9434b7d5cf529045d831ce904cf38d")
>>> b.resources_manager.icons
Segmentation fault (core dumped)
```

```
Python 3.9.7 (default, Sep 16 2021, 13:09:58)
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import lief
>>> lief.__version__
'0.12.0-e0aea95'
>>> b = lief.parse("22e4217a21311b5af94a166366395209ac9434b7d5cf529045d831ce904cf38d")
>>> b.resources_manager.icons
Segmentation fault (core dumped)
```

**Expected behavior**
LIEF usually throws an error (lief.corrupted, lief.read_out_of_bound, ...) when the resources are too heavily corrupted and unparsable. Such an error would be best.

**Environment**
 - System and Version : Ubuntu 18.04
 - Target format : PE
 - Target binary : [22e4217a21311b5af94a166366395209ac9434b7d5cf529045d831ce904cf38d](https://www.virustotal.com/gui/file/22e4217a21311b5af94a166366395209ac9434b7d5cf529045d831ce904cf38d)
 - LIEF commit version: Tested with 0.11.5-37bc2c9 and 0.12.0-e0aea95

**Comment by romainthomas** [comment]
Ack, could you give me the output of ``b.has_resources`` ?

**Comment by gdesmar** [comment]
```python
>>> import lief
>>> lief.__version__
'0.11.5-37bc2c9'
>>> b = lief.parse("22e4217a21311b5af94a166366395209ac9434b7d5cf529045d831ce904cf38d")
>>> b.has_resources
True
```

Same result with the nightly version.

**Comment by romainthomas** [comment]
Ok, could you attach the sample? (I don't have access to VT)

**Comment by gdesmar** [comment]
You can download it from the following site : sorel-20m.s3.amazonaws.com/09-DEC-2020/binaries/22e4217a21311b5af94a166366395209ac9434b7d5cf529045d831ce904cf38d 
The malware is compressed via zlib and disarmed by having both the OptionalHeader.Subsystem flag and the FileHeader.Machine header value set to 0. You don't need to find the right values and re-arm the malware to trigger the problem, so the following line of code should be good enough, if you have zlib-flate.
```bash
wget -O- sorel-20m.s3.amazonaws.com/09-DEC-2020/binaries/22e4217a21311b5af94a166366395209ac9434b7d5cf529045d831ce904cf38d | zlib-flate -uncompress > 22e4217a21311b5af94a166366395209ac9434b7d5cf529045d831ce904cf38d
```

**Comment by romainthomas** [comment]
Thanks for the report I fixed it through  d76018c
